docs/DeveloperGuide.md

```diff
@@ -1,38 +1,146 @@
+3. 
+
+The sequence diagram below shows how this feature works:
+{UML will be added here.}
+
+
+### Tag Feature
+
+### Export Feature
+
+### Deck Feature
+
+---
+## Appendix: Requirements
+
+### Product scope
 
-## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+- Person who wants a convenient, versatile and portable way to revise concepts while doing coding assignments
+- Person who wants to effectively learn a new language while doing coding assignments
+- Person who wants to be challenged to come up with answers quickly against time pressure.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+User will be able to revise while coding or using the terminal so they don&#39;t have to switch between apps. This will maximize their productivity and refresh their memory instantly whenever they want to recall some topics suddenly appear on their mind.
+
 
-## User Stories
+### User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...                                     | So that I could ...                                               |     
+|---------|----------|---------------------------------------------------|-------------------------------------------------------------------|
+|  v1.0   | new user	| add a card containing the questions and answers	 | store the question in the system for later revision               |
+|  v1.0	 | new user	| see a list of all cards	                         | I can see what are all the cards I have regardless of their tags  |
+|  v1.0   |	new user	| delete some cards	                               | I can remove unwanted cards                                       |
+|  v1.0	 | new user	| save cards to a local file	                      | I can reuse the cards the next time I use the software            |
+|  v1.0	 | new user	| load questions and answers from a file	          | I can use the database of flashcards                              |
+|  v1.0	 | new user	| save cards to a local file	                      | I can reuse the cards the next time I use the software            |
+|  v1.0	 | new user	| load questions and answers from a file	          | I can use the database of flashcards                              |
+|  v2.0	 | new user	| update the content of the card	                   | I can ensure that the content of the card is up-to-date           |
+|  v2.0	 | new user	| group my cards according to my will into "Deck"	 | I can easily access all of my cards in a certain category         |
+|  v2.0	 | new user	| tag each card into one or more topics	          | I can organize my cards according to the topics associated        |  
```

Comment:
It seems like the tagging feature is something that may not be best suited for a new user, but instead more useful for an existing/advanced user.